fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -289,7 +327,8 @@ public void deleteFile(@Nonnull String name) throws IOException {
                     this.fileReferenceCache.invalidate(name);
                     return true;
                 }
-        ).join();
+        ),start).join();
```

Comment:
You shouldn't be using `join()` -- pretty much anywhere.  This is why we have `WAIT` timers.  So the code, should be something like:

    boolean deleted = context.asyncToSync(FDBStoreTimer.Waits.WAIT_LUCENE_DELETE_FILE, 
        getFDBLuceneFileReference(name).thenApply(.... ));

`join()` can block indefinitely, and the `asyncToSync` infrastructure buys you a lot of error handling and other benefits.